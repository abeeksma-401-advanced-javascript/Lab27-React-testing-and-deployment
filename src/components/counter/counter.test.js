'use strict';

import React from 'react';
import renderer from 'react-test-renderer';

import Counter from './counter';


describe('The Count(er)', () => {
  it('renders shallowly like fucking life', () => {
    let counter = shallow(<Counter/>);
    expect(counter.find('section').exists()).toBe(true);
  });

  it('mounts without props and does counting', () => {
    let counter = mount(<Counter/>);
    let buttonUp = counter.find('.down');
    let buttonDown = counter.find('.up');
    let countText = counter.find('span')

    expect(buttonUp.exists()).toBe(true);
    expect(buttonDown.exists()).toBe(true);
    
    expect(counter.find('span').exists()).toBe(true);

    buttonUp.simulate('click');
    expect(countText.text()).toBe('-1');

    buttonDown.simulate('click');
    expect(countText.text()).toBe('0');

  });
});