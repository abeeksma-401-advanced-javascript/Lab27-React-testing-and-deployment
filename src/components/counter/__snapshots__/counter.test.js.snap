// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Count(er) matches snapshot without props 1`] = `
<section
  className="counter"
>
  <a
    className="down clicker"
    href="#"
    onClick={[Function]}
  >
    -
  </a>
  <span
    className="count "
  >
    0
  </span>
  <a
    className="up clicker"
    href="#"
    onClick={[Function]}
  >
    +
  </a>
</section>
`;
